### PR TITLE
[eas-cli] Limit the number of SignedAssetUploadSpecifications fetched at a time

### PR DESCRIPTION
**Pull request by wschurman**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

Generating a ton of these at once on the server creates 1) requests that are huge and that have huge responses, and 2) a lot of load to google to generate the specs. Both of these issues were fixed on the server in https://github.com/expo/universe/pull/10394, but we still need to do batching on the client in order to not get timeouts. 

# How

I think doing 100 at a time is fine. For 3k assets, this took ~10 seconds or so to get all the specifications when I tested locally.

# Test Plan

For an app with 3k assets:
```
export EXPO_STAGING=1
~/expo/eas-cli/packages/eas-cli/bin/run update --branch preview --message "Updating the app"
```
See that it works!!

### CODE REVIEW

**Comment by wschurman** [comment]
/changelog-entry bug-fix Limit the number of SignedAssetUploadSpecifications fetched at a time

File: packages/eas-cli/src/project/publish.ts [hunk 1]
```diff
@@ -285,12 +286,18 @@ export async function uploadAssetsAsync(
...
 
   updateSpinnerText?.(totalAssets, totalAssets);
-
   let missingAssets = await filterOutAssetsThatAlreadyExistAsync(uniqueAssets);
   const uniqueUploadedAssetCount = missingAssets.length;
-  const { specifications } = await PublishMutation.getUploadURLsAsync(
-    missingAssets.map(ma => ma.contentType)
-  );
+
+  const missingAssetChunks = chunk(missingAssets, 100);
```
Comment (ide) [inline packages/eas-cli/src/project/publish.ts:292]: How well do we handle 1000?

**Review by ide — APPROVED** [review]

**Review by dsokal — APPROVED** [review]

File: packages/eas-cli/src/project/publish.ts (hunk 1 above)
Comment (wschurman) [reply packages/eas-cli/src/project/publish.ts:292]: We handle 1400 in prod relatively ok (haven't seen failures), 100 is more just setting this to a value that doesn't make this too slow for normal apps (1-2 requests for most apps I assume) while keeping us far away from hitting the request timeout.

Comment (ide) [reply packages/eas-cli/src/project/publish.ts:292]: OK. My thinking is that we want to find a sweet spot that uses batching efficiently while also not overloading a replica and causing it to run for too long (~1 second would be a high target IMO).

**Pull request merged**